test/metabase/data_apps/api/data_app_test.clj

```diff
@@ -0,0 +1,130 @@
+(ns metabase.data-apps.api.data-app-test
+  "Tests for /api/data-app endpoints"
+  (:require
+   [clojure.test :refer :all]
+   [metabase.data-apps.test-util :as data-apps.tu]
+   [metabase.test :as mt]
+   [toucan2.core :as t2]))
+
+(deftest comprehensive-crud-flow-test
+  (testing "Complete CRUD flow for data apps including definition and release"
+    (data-apps.tu/with-data-app-cleanup!
+      (mt/with-test-user :crowberto
+        (let [create-response (mt/user-http-request :crowberto :post 200 (data-apps.tu/data-app-url)
+                                                    {:name        "My Data App"
+                                                     :slug        "/my-data-app"
+                                                     :description "A comprehensive test app"})
+              app-id (:id create-response)]
+          (testing "Create returns expected fields"
+            (is (=? {:id          (mt/malli=? pos-int?)
+                     :name        "My Data App"
+                     :slug        "/my-data-app"
+                     :description "A comprehensive test app"
+                     :creator_id  (mt/user->id :crowberto)}
+                    create-response)))
+
+          (testing "Update app fields"
+            (let [update-response (mt/user-http-request :crowberto :put 200 (data-apps.tu/data-app-url app-id)
+                                                        {:name        "Updated Data App"
+                                                         :slug        "/updated-data-app"
+                                                         :description "Updated description"})]
+              (is (=? {:name        "Updated Data App"
+                       :slug        "/updated-data-app"
+                       :description "Updated description"}
+                      update-response))))
+
+          (testing "Add definition to app"
+            (let [definition {:config data-apps.tu/default-app-definition-config}
```

Comment:
Hmm feels weird to wrap the definition in `:config` here. Are you protecting against us needing to add other keys later?

I would just send it naked for now, or if you need to wrap it just use the resource name `{:definition ...}` (and wrap it inline in the `user-http-request` call).

Reply:
Decided to leave as `:config` for now, easy to revisit.